spec/models/results/response_tree_builder_spec.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe "response tree" do
+  # form with repeat group
+  context "simple form" do
+    let(:form1) { create(:form, question_types: %w[text text text]) }
+
+    it "should produce a simple tree from a form with three children" do
+      response_tree = Results::ResponseTreeBuilder.new(form1).build
+      # same number of leaves at the root
+      expect(response_tree.root_answers.count).to eq form1.root_questionings.count
+
+      # expect to be the same qing_ids
+      expect(response_tree.root_answers.map(&:questioning_id))
+        .to include(form1.root_questionings[0].id, form1.root_questionings[1].id,
+          form1.root_questionings[2].id)
+
+      # same ranks
+      expect(response_tree.root_answers.map(&:new_rank))
+        .to include(form1.root_questionings[0].rank, form1.root_questionings[1].rank,
+          form1.root_questionings[2].rank)
+    end
+  end
+
+  context "forms with groups" do
+    let(:form2) { create(:form, question_types: ["text", %w[select_one text], "text"]) }
```

Comment:
You should just do `let(:form)` throughout. No need to add different suffixes. In fact it's better not to, b/c then the `let(:form_tree)` thing can live **at the top level**!! 💥